maven-core/src/main/java/org/apache/maven/internal/transformation/TransformedArtifact.java

```diff
@@ -67,36 +71,36 @@ public void setFile(File file) {
 
     @Override
     public File getFile() {
-        Path result = onChangeTransformer.get();
-        if (result == null) {
-            return null;
+        try {
+            Path result = onChangeTransformer.transform();
+            if (result == null) {
+                return null;
+            }
+            return result.toFile();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
Never throw raw runtime exceptions. Even if the overridden API doesn't allow checked exceptions, use a more specific subclass of RuntimeException, though in this case it might be better to return null